test/run

```diff
@@ -2,8 +2,11 @@
 
 set -eu
 
-# run static code checks like pyflakes and pep8
-PYEXEFILES="$(git grep -lI '^#!.*python') $(git ls-files "*.py")"
-flake8 $PYEXEFILES
+test_ruff() {
+    test -n "$(ruff check --show-files . 2>/dev/null)" || return 0  # no python changes
```

Comment:
This will silently ignore a failing or non-existing `ruff`, which may be your intention. But at least showing some "skip: ruff not installed" would be nice?